addon/components/sortable-group.js

```diff
@@ -163,10 +187,202 @@ export default Component.extend({
       });
     });
 
-    if (groupModel !== NO_MODEL) {
-      this.onChange(groupModel, itemModels, draggedModel);
-    } else {
+    set(this, 'modelCopy', a(itemModels));
+    // if (groupModel !== NO_MODEL) {
+    //   this.onChange(groupModel, itemModels, draggedModel);
+    // } else {
       this.onChange(itemModels, draggedModel);
+    // }
+  },
+
+  /**
+   * Explanation
+   * 1. `KeyboardReorderMode` is disabled: users can activate it via ENTER or SPACE.
+   * 2. `KeyboardReorderMode` is enabled: users can reorder via UP or DOWN arrow keys. TODO: Expand to more keys, e.g LEFT, RIGHT
+   * 3. `KeyboardReorderMode` is enabled: users can finalize/save the reordering via ENTER or SPACE.
+   * 4. `KeyboardReorderMode` is enabled: users can discard the reordering via ESC.
+   *
+   * @param {Event} evt a HTML event
+   */
+  keyDown(event) {
+    // Note: If handle is specified, we need to target the keyDown on the handle
+    const  { isKeyboardReorderModeEnabled, direction, selectedItem, sortedItems } = this.getProperties('isKeyboardReorderModeEnabled', 'direction', 'selectedItem', 'sortedItems');
+
+    if (!isKeyboardReorderModeEnabled && (isEnterKey(event) || isSpaceKey(event))) {
+      this.prepareKeyboardReorderMode();
+      this.set('isRetainingFocus', true);
+      readDOM(() => {
+        this.element.focus();
+        this.set('isRetainingFocus', false);
+      });
+      // Prevent the default scroll
+      event.preventDefault();
+      return;
+    }
+
+    if (isKeyboardReorderModeEnabled) {
+      if (direction === "y" && isDownArrowKey(event)) {
+        // Guarantees that the before the UI is fully rendered before we move again.
+        mutateDOM(() => {
+          this.moveItem(selectedItem, 1);
+        })
+      } else if (direction === "y" && isUpArrowKey(event)) {
+        mutateDOM(() => {
+          this.moveItem(selectedItem, -1);
+        })
+      } else if (direction === "x" && isLeftArrowKey(event)) {
+        mutateDOM(() => {
+          this.moveItem(selectedItem, -1);
+        })
+      } else if (direction === "x" && isRightArrowKey(event)) {
+        mutateDOM(() => {
+          this.moveItem(selectedItem, 1);
+        })
+      } else if (isEnterKey(event) || isSpaceKey(event)) {
+        this.confirmKeyboardSelection();
+
+        readDOM(() => {
+          const itemElement = this.selectedItem.element;
+          const handle = itemElement.querySelector('[data-sortable-handle]');
+          if (handle) {
+            handle.focus();
+          } else {
+            // The consumer did not use a handle, so we set focus back to the item.
+            itemElement.focus();
+          }
+        });
+      } else if (isEscapeKey(event)) {
+        this.cancelKeyboardSelection();
+        this.set('isRetainingFocus', true);
+        readDOM(() => {
```

Comment:
it will be nice to call it as different method as well, with ability to cancel it if we reenter here.

Reply:
👍

Reply:
I decided to abstract the `focus` logic, but not the entire logic because it seemed a bit unnecessary imo. we have `cancelKeyboard` and then focus logic, which should be readable.